### ISSUE TITLE
Cannot connect to ArangoDB when root password is required

### ISSUE THREAD

**Issue by kycheng801** [roboncode/orango#64]
**Version:** 1.0.0-alpha-2.1
**Arango:** arangodb:3.4.2

**Code:**

```javascript
async function initDatabase() {
  await orango.connect({
    url: config.db.url,
    username: config.db.rootUser,
    password: config.db.rootPass
  });
  await orango.dropDatabase(config.db.name);
  await orango.createDatabase(config.db.name);
  await orango.disconnect();
}

async function main() {
  await initDatabase();

  let db = orango.get(config.db.name);

  db.events.once(EVENTS.CONNECTED, conn => {
    console.log('🥑  Connected to ArangoDB:'.green, conn.url + '/' + conn.name)
  });

  db.events.once(EVENTS.READY, () => {
    console.log('🍊  Orango is ready!'.green)
  });

  await db.connect();
}
```

**Expected:** A new database called `config.db.name` is created successfully.

**Actually:**

```console
$ node scripts/test.js
(node:1895) UnhandledPromiseRejectionWarning: ArangoError: not authorized to execute this request
    at new ArangoError (/Users/kennycheng/workspace/belstar/odm/apps/_db/_system/core/core-service/node_modules/arangojs/src/error.ts:76:17)
    at Object.resolve (/Users/kennycheng/workspace/belstar/odm/apps/_db/_system/core/core-service/node_modules/arangojs/src/connection.ts:350:20)
    at _hosts.(anonymous function) (/Users/kennycheng/workspace/belstar/odm/apps/_db/_system/core/core-service/node_modules/arangojs/src/connection.ts:209:16)
    at IncomingMessage.res.on (/Users/kennycheng/workspace/belstar/odm/apps/_db/_system/core/core-service/node_modules/arangojs/src/util/request.node.ts:117:15)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
(node:1895) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1895) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

After tracing the source code, I found the problem should be here:

<img width="537" alt="2019-02-15 10 50 18" src="https://user-images.githubusercontent.com/6451053/52831536-f2d7a500-310f-11e9-9cfb-9f0ef3d56e7b.png">

If the default root password has been set, the `exists` check will definitely fail due to not yet login. Maybe just letting Arango return an error code is a good enough option?

**Comment by roboncode** [comment]
Ty. Looking into it.

**Comment by roboncode** [comment]
Please review at https://github.com/roboncode/orango/tree/bugfix/64, I believe this will fix this issue. I have refactored the connection functionality in orango and this fix should address the bug you were seeing.

**Comment by roboncode** [comment]
This issue has been fixed with the release of v1.0.0-alpha-2.2
